src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -175,17 +158,42 @@ class MarathonSchedulerActor private (
       removeLocks(plan.affectedRunSpecIds)
       deploymentFailed(plan, reason)
 
-    case RunSpecScaled(id) => removeLock(id)
-
     case TasksKilled(runSpecId, _) => removeLock(runSpecId)
 
+    case StartInstance(runSpec) =>
+      if (noConflictsWith(Set(runSpec.id))) startInstance(runSpec)
+      else logger.info(s"Did not start an instance for ${runSpec.id} because runSpec is locked in a deployment.")
+
+    case DecommissionInstance(runSpec) =>
+      if (noConflictsWith(Set(runSpec.id))) decommissionInstance(runSpec)
+      else logger.info(s"Did not decommission an instance for ${runSpec.id} because runSpec is locked in a deployment.")
+
     case msg => logger.warn(s"Received unexpected message from ${sender()}: $msg")
   }
 
-  def scaleRunSpecs(): Unit = {
-    groupRepository.root().foreach { root =>
-      root.transitiveRunSpecs.foreach(spec => self ! ScaleRunSpec(spec.id))
+  private def startInstance(runSpec: RunSpec): Unit = {
+    launchQueue.add(runSpec, 1)
```

Comment:
Since we want to get rid off the launch queue I'd schedule or reschedule the instance explicitly:

```suggestion
async {
  await(instanceTracker.specInstances(runSpec.id)).find(_.state.goal == Goal.Stopped) match {
    case Some(instance) =>
      instanceTracker.process(RescheduleReserved(instance.instanceId, runSpec))
    case None =>
      instanceTracker.schedule(Instance.scheduled(runSpec))
  }
  Done
}
```

what do you think?